From cf7ca7885edd10acca5fdadb2057bcb2a7b05ae5 Mon Sep 17 00:00:00 2001
From: Lin Ruizhe <linruizhe@huawei.com>
Date: Fri, 31 Dec 2021 15:59:26 +0800
Subject: [PATCH] msi: Add kabi_reserve in msi.h
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cf7ca7885edd10acca5fdadb2057bcb2a7b05ae5


hulk inclusion
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MZU1
CVE: NA

---------------------------

Add KABI_RESERVE in msi.h

Signed-off-by: Lin Ruizhe <linruizhe@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/msi.h | 3 +++
 1 file changed, 3 insertions(+)

--- a/include/linux/msi.h
+++ b/include/linux/msi.h
@@ -4,6 +4,7 @@
 
 #include <linux/kobject.h>
 #include <linux/list.h>
+#include <linux/kabi.h>
 #include <asm/msi.h>
 
 /* Dummy shadow structures if an architecture does not define them */
@@ -167,6 +168,8 @@ struct msi_desc {
 		struct fsl_mc_msi_desc fsl_mc;
 		struct ti_sci_inta_msi_desc inta;
 	};
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
 };
 
 /* Helpers to hide struct msi_desc implementation details */
